### ISSUE TITLE
[BUG]: Models ArcFace, Dlib, and others are not working. VGG-Face is the only model working for me

### ISSUE THREAD

**Issue by arkadaw9** [serengil/deepface#1218]
### Before You Report a Bug, Please Confirm You Have Done The Following...

- [X] I have updated to the latest version of the packages.
- [X] I have searched for both [existing issues](https://github.com/serengil/deepface/issues) and [closed issues](https://github.com/serengil/deepface/issues?q=is%3Aissue+is%3Aclosed) and found none that matched my issue.

### DeepFace's version

v0.0.90

### Python version

3.9.19

### Operating System

Ubuntu 20.04

### Dependencies

Some packages versions I am currently using.

# Name                    Version                   Build  Channel
cuda-version              11.8                 h70ddcb2_3    conda-forge
cudatoolkit               11.8.0              h4ba93d1_13    conda-forge
cudnn                     8.9.7.29             hbc23b4c_3    conda-forge
deepface                  0.0.90                   pypi_0    pypi
distributed               2023.10.1          pyhd8ed1ab_0    conda-forge
dlib                      19.24.4         cuda118py39ha3e66a0_0    conda-forge
easydict                  1.9                        py_0    conda-forge
expat                     2.6.2                h59595ed_0    conda-forge
facemodels                0.1.0rc0                 pypi_0    pypi
facenet-pytorch           2.5.3                    pypi_0    pypi
ffmpeg                    6.1.1           gpl_h8007c5b_104    conda-forge
fsspec                    2024.3.1           pyhca7485f_0    conda-forge
gdown                     5.1.0                    pypi_0    pypi
h5py                      3.10.0          nompi_py39h2c511df_101    conda-forge
hdf5                      1.14.3          nompi_h4f84152_100    conda-forge
imagecodecs               2024.1.1         py39hd694e17_2    conda-forge
imageio                   2.34.0             pyh4b66e23_0    conda-forge
insightface               0.1.5                    pypi_0    pypi
isoduration               20.11.0                  pypi_0    pypi
itsdangerous              2.2.0                    pypi_0    pypi
jasper                    4.2.2                he6dfbbe_0    conda-forge
jax                       0.4.26                   pypi_0    pypi
jaxlib                    0.4.26                   pypi_0    pypi
json5                     0.9.24                   pypi_0    pypi
jsonpointer               2.4                      pypi_0    pypi
jsonschema                4.21.1             pyhd8ed1ab_0    conda-forge
keras                     3.2.1                    pypi_0    pypi
matplotlib                3.8.3            py39hf3d152e_0    conda-forge
mediapipe                 0.10.11                  pypi_0    pypi
mtcnn                     0.1.1                    pypi_0    pypi
mxnet                     1.9.1                    pypi_0    pypi
numpy                     1.23.5                   pypi_0    pypi
numpy-base                1.26.4           py39h8a23956_0  
onnx                      1.15.0           py39h868dac8_0    conda-forge
onnx2torch                1.5.12                   pypi_0    pypi
onnxruntime               1.17.1          py39h465c7e1_202_cuda    conda-forge
opencv                    4.9.0           qt5_py39h8123b4e_510    conda-forge
opencv-contrib-python     4.9.0.80                 pypi_0    pypi
pandas                    2.2.1            py39hddac248_0    conda-forge
pillow                    10.2.0           py39had0adad_0    conda-forge
pip                       24.0               pyhd8ed1ab_0    conda-forge
pytorch                   2.1.0           cuda112_py39h53f755c_303    conda-forge
pytorch-gpu               2.1.0           cuda112py39h398211c_303    conda-forge
pytorch-lightning         2.2.1              pyhd8ed1ab_0    conda-forge
pytz                      2024.1             pyhd8ed1ab_0    conda-forge
pyu2f                     0.1.5              pyhd8ed1ab_0    conda-forge
pywavelets                1.4.1            py39h44dd56e_1    conda-forge
requests                  2.31.0             pyhd8ed1ab_0    conda-forge
retina-face               0.0.17                   pypi_0    pypi
scikit-image              0.22.0           py39hddac248_2    conda-forge
scikit-learn              1.1.3                    pypi_0    pypi
scipy                     1.12.0           py39h474f0d3_2    conda-forge
seaborn                   0.13.2                   pypi_0    pypi
tensorboard               2.16.2                   pypi_0    pypi
tensorboard-data-server   0.7.0            py39hd4f0224_1    conda-forge
tensorboardx              2.6.2.2            pyhd8ed1ab_0    conda-forge
tensorflow                2.16.1                   pypi_0    pypi
tensorflow-cpu            2.15.0          cpu_py39h718b53a_2    conda-forge
tensorflow-estimator      2.12.0                   pypi_0    pypi
tensorflow-io-gcs-filesystem 0.36.0                   pypi_0    pypi
tf-keras                  2.16.0                   pypi_0    pypi
torchmetrics              1.3.2              pyhd8ed1ab_0    conda-forge
torchvision               0.16.1          cuda112py39h4572277_2    conda-forge
tqdm                      4.66.2             pyhd8ed1ab_0    conda-forge

### Reproducible example

```Python
from sklearn.datasets import fetch_lfw_pairs
from sklearn.metrics import accuracy_score, precision_score, recall_score, f1_score, confusion_matrix
import matplotlib.pyplot as plt
from deepface import DeepFace
from tqdm import tqdm
import pandas as pd

#load LFW data
fetch_lfw_pairs = fetch_lfw_pairs(subset = 'test', color = True, resize = 1)

pairs = fetch_lfw_pairs.pairs
labels = fetch_lfw_pairs.target
target_names = fetch_lfw_pairs.target_names

instances = pairs.shape[0]
print("instances: ", instances)

# Evaluate results
actuals = []; predictions = []; distances = []
pbar = tqdm(range(0, instances))

for i in pbar:
    pair = pairs[i]
    img1 = pair[0]; img2 = pair[1]
    img1 = img1[:,:,::-1]; img2 = img2[:,:,::-1] #opencv expects bgr instead of rgb
    
    #obj = DeepFace.verify(img1, img2, model_name = 'VGG-Face', model = vgg_model)
    obj = DeepFace.verify(img1, img2, model_name = 'Dlib', distance_metric = 'euclidean', enforce_detection=False)
    # obj = DeepFace.verify(img1, img2, model_name = 'ArcFace', distance_metric = 'cosine', enforce_detection=False, normalization="ArcFace")
    prediction = obj["verified"]
    predictions.append(prediction)
    
    distances.append(obj["distance"])
    
    label = target_names[labels[i]]
    actual = True if labels[i] == 1 else False
    actuals.append(actual)

accuracy = 100*accuracy_score(actuals, predictions)
precision = 100*precision_score(actuals, predictions)
recall = 100*recall_score(actuals, predictions)
f1 = 100*f1_score(actuals, predictions)

print("instances: ",len(actuals))
print("accuracy: " , accuracy, "%")
print("precision: ", precision, "%")
print("recall: ", recall,"%")
print("f1: ",f1,"%")
```

### Relevant Log Output

**Log Output for Model Dlib:**
instances:  1000
100%|███████████████████████████████████████████████████████████████| 1000/1000 [00:07<00:00, 128.84it/s]
instances:  1000
accuracy:  50.0 %
precision:  50.0 %
recall:  100.0 %
f1:  66.66666666666666 %

**Log Output for Model ArcFace:**
2024-04-26 11:14:20.836879: I tensorflow/core/platform/cpu_feature_guard.cc:210] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 AVX512F FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
2024-04-26 11:14:21.529540: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT
instances:  1000
  0%|                                                                           | 0/1000 [00:00<?, ?it/s]2024-04-26 11:14:23.486832: W tensorflow/core/common_runtime/gpu/gpu_device.cc:2251] Cannot dlopen some GPU libraries. Please make sure the missing libraries mentioned above are installed properly if you would like to use GPU. Follow the guide at https://www.tensorflow.org/install/gpu for how to download and setup the required libraries for your platform.
Skipping registering GPU devices...
100%|████████████████████████████████████████████████████████████████| 1000/1000 [05:18<00:00,  3.14it/s]
instances:  1000
accuracy:  50.0 %
precision:  50.0 %
recall:  100.0 %
f1:  66.66666666666666 %

### Expected Result

The code example is from your [LFW Tutorial](https://sefiks.com/2020/08/27/labeled-faces-in-the-wild-for-face-recognition/). I was expecting the results to be similar to the ones reported (or atleast greater than 95%). 

### What happened instead?

The model performances are very poor around 50% even after finetuning the thresholds. Can you help me identify what is going wrong?

**Comment by serengil** [comment]
you are referencing an external study as a bug. deepface's issue tracker is not right place to report this.

try to save the coming image from lfw to your file system then pass the image's path to deepface.

**Comment by serengil** [comment]
you should ask your questions on stackoverflow as mentioned in issue templates.

Do you know why saving the images to a file and passing the image path might work differently? > Most probably sklearn stores them as compressed. Once they are saved, DeepFace can use them correctly.
